invenio_records_resources/services/custom_fields/base.py

```diff
@@ -13,12 +13,25 @@
 class BaseCF(ABC):
     """Base Custom Field class."""
 
-    def __init__(self, name):
+    def __init__(self, name, required=False, validate=None, error_messages=None):
         """Constructor."""
         self.name = name
         self.relation_cls = None
```

Comment:
I think this can be moved in another base class eg: `BaseRelationCF` where all the custom fields related to another class i.e Vocabulary can implement their specific API...wdyt?

Reply:
While I fully agree, it is not possible right now because that attribute is used in many parts of the custom fields implementation to differentiate normal CF from Vocabulary CFs. Solving it, is a wider topic that will be tackled here <https://github.com/inveniosoftware/invenio-records-resources/issues/358>